lib/transaction.js

```diff
@@ -180,6 +188,12 @@ Transaction.prototype.prepareEnvironment = function() {
     return self.setIsolationLevel();
   }).then(function () {
     return self.setAutocommit();
+  }).catch(function (err) {
+    try {
+      self.rollback();
+    } catch (e) {}
+
+    throw err;
```

Comment:
Well in case `rollback` does in fact execute, it's best that we actually wait for it to finish before proceeding :)